diff --git a/node_modules/docusaurus-theme-openapi-docs/lib/theme/Schema/index.js b/node_modules/docusaurus-theme-openapi-docs/lib/theme/Schema/index.js
--- a/node_modules/docusaurus-theme-openapi-docs/lib/theme/Schema/index.js
+++ b/node_modules/docusaurus-theme-openapi-docs/lib/theme/Schema/index.js
@@ -95,6 +95,9 @@
 };
 const Summary = ({ name, schemaName, schema, required }) => {
   const { deprecated, nullable } = schema;
+  const schemaBadges = Array.isArray(schema["x-badges"])
+    ? schema["x-badges"].filter((b) => b && b.name)
+    : [];
   const isRequired = Array.isArray(required)
     ? required.includes(name)
     : required === true;
@@ -119,10 +122,26 @@
         " ",
         schemaName
       ),
-      (isRequired || deprecated || nullable) &&
+      (isRequired || deprecated || nullable || schemaBadges.length > 0) &&
         react_1.default.createElement("span", {
           className: "openapi-schema__divider",
         }),
+      schemaBadges.map((b, i) =>
+        react_1.default.createElement(
+          "span",
+          {
+            key: i,
+            className: (0, clsx_1.default)(
+              "openapi-schema__badge",
+              `openapi-schema__badge--${(b.name || "")
+                .toLowerCase()
+                .trim()
+                .replace(/\s+/g, "-")}`
+            ),
+          },
+          b.name
+        )
+      ),
       nullable &&
         react_1.default.createElement(
           "span",
